android/src/main/java/io/jari/fingerprint/FingerprintModule.java

```diff
@@ -14,171 +13,125 @@
-        }
+    @SuppressWarnings("unused")
+    public final void hasPermission(final Promise promise) {
+        promise.resolve(
+                ActivityCompat.checkSelfPermission(
+                        getReactApplicationContext(),
+                        Manifest.permission.USE_FINGERPRINT
+                ) == PackageManager.PERMISSION_GRANTED
+        );
     }
 
     @ReactMethod
-    public void hasEnrolledFingerprints(Promise promise) {
-        try {
-            promise.resolve(fingerprintManager.hasEnrolledFingerprints());
-        } catch (SecurityException secEx) {
-            Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
-            exception.initCause(secEx);
-            promise.reject(exception);
-        } catch (Exception ex) {
-            promise.reject(ex);
-        }
+    @SuppressWarnings("unused")
+    public final void hasEnrolledFingerprints(final Promise promise) {
+        promise.resolve(fpm.hasEnrolledFingerprints());
     }
 
     @ReactMethod
-    public void isHardwareDetected(Promise promise) {
-        try {
-            promise.resolve(fingerprintManager.isHardwareDetected());
-        } catch (SecurityException secEx) {
-            Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
-            exception.initCause(secEx);
-            promise.reject(exception);
-        } catch (Exception ex) {
-            promise.reject(ex);
-        }
+    @SuppressWarnings("unused")
+    public final void isHardwareDetected(final Promise promise) {
+        promise.resolve(fpm.isHardwareDetected());
     }
 
     @ReactMethod
-    public void authenticate(Promise promise) {
+    @SuppressWarnings("unused")
+    public final void authenticate(final Promise promise) {
         try {
-            isCancelled = false;
-            cancellationSignal = new CancellationSignal();
-            fingerprintManager.authenticate(null, cancellationSignal, 0, new AuthenticationCallback(promise), null);
-        } catch (SecurityException secEx) {
-            Exception exception = new Exception("App does not have the proper permissions. (did you add USE_FINGERPRINT to your manifest?)\nMore info see https://github.com/jariz/react-native-fingerprint-android");
-            exception.initCause(secEx);
-            promise.reject(exception);
-        } catch (Exception ex) {
-            promise.reject(ex);
+            IS_CANCELED = false;
+            CANCELLATION_SIGNAL = new CancellationSignal();
+            fpm.authenticate(
+                    null,
+                    0,
+                    CANCELLATION_SIGNAL,
+                    new FingerprintManagerCompat.AuthenticationCallback() {
```

Comment:
Why inline this? Why not keep it separate like it was before. Kind of ugly this way.